pyproject.toml

```diff
@@ -0,0 +1,54 @@
+[project]
+name = "stactools-met-office-deterministic"
+version = "0.1.0"
+description = "Python package for generating STAC metadata for the Met Office Deterministic Numerical Weather Prediction model"
+readme = "README.md"
+authors = [
+    { name = "hrodmn", email = "henry@developmentseed.org" }
+]
+keywords = ["stactools", "pystac", "catalog", "STAC"]
+license = { text = "MIT" }
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
+]
+requires-python = ">=3.11"
+dependencies = [
+    "geojson-pydantic>=2.1.0",
+    "obstore>=0.8.2",
+    "pystac>=1.14.1",
+]
+
+
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/stactools"]
+
+[dependency-groups]
+dev = [
+    "codespell==2.4.1",
+    "ipython>=9.6.0",
+    "mypy==1.18.2",
+    "pre-commit>=4.3.0",
+    "ruff==0.14.1",
```

Comment:
It'd be good to relax the **mypy** and **ruff** dependencies to `>=`, and let `uv.lock` handle the pinning.

Reply:
Got it - I added `sync-with-uv` to make sure pre-commit and uv.lock stay in sync :smile: